### DISCUSSION TITLE
Strange behavior for conditional serialized jobs

### DISCUSSION THREAD

**Discussion by jmalak** [community/community#26995] [Actions]
<p>I tried to use serialized jobs with conditions.</p>
<p>I supposed that condition should be applied only for one jobs, but it looks like it is applied to all serialized jobs that it behaves as one job. If any condition is false it finishs workflow.</p>
<p>I have a test Workflow when it is demonstrated</p>
<p><a href="https://github.com/jmalak/test/actions/runs/99224343" rel="nofollow noopener">https://github.com/jmalak/test/actions/runs/99224343</a></p>
<p>Is it possible to ensure continue of next job in thread somehow?</p>
<p>I need to trigger workflow by change in some files and filter trigger events also by branch.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jmalak">@jmalak</a> ,</p>
<p>You can try the below configurations on  <strong>job4</strong> , this should work as expected.</p>
<pre><code>job4:
    runs-on: ubuntu-latest
    needs: job3
    if: github.ref == 'refs/heads/master' &amp;&amp; always()
    steps:
    - name: job4
      run: echo "job4"
</code></pre>

**Comment by jmalak** [comment]
<p><a class="mention" href="/brightran">@brightran</a> Thanks, it works as expected.</p>
<p>When I want to run it always even if previous job was not run due to condition is false then I must add</p>
<pre><code>if: always()
</code></pre>
<p>condition for such job.</p>
<p>It looks like Workflow has internal condition flag which is global to thread and therefore it is changed by evaluation of each condition.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/jmalak">@jmalak</a> ,</p>
<p>Yeah, currently, if we want a job continues to run when its <strong>needs</strong> job is failed/canceled/skipped, we seems must use the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions" rel="nofollow noopener"><strong>Job status check functions</strong></a> in the <strong>if conditionals</strong> of this job.</p>

**Comment by burbigo1** [comment]
<p>Thanks a lot for sharing the solution with us</p>
